crates/sui-config/src/genesis.rs

```diff
@@ -979,11 +984,10 @@ fn process_package(
     vm: &MoveVM,
     ctx: &mut TxContext,
     modules: Vec<CompiledModule>,
+    dependencies: Vec<ObjectID>,
     protocol_config: &ProtocolConfig,
 ) -> Result<()> {
-    let inputs = Transaction::input_objects_in_compiled_modules(&modules);
-    let ids: Vec<_> = inputs.iter().map(|kind| kind.object_id()).collect();
-    let input_objects = store.get_objects(&ids[..]);
+    let dependency_objects = store.get_objects(&dependencies[..]);
```

Comment:
Love the simplification ❤️

minor nit(?): Can the slicing be removed as well here? so just `&dependencies`?

Reply:
Yeah possibly!  I'll try and see -- in another place in this file, Rust struggled to coerce `&[T; n]` to `&[T]` so I wonder if explicit slice conversion was added to counteract that.  I can't think of a reason why it's currently needed though.

Reply:
(Yep, got rid of it!)